service/model/project_membership.go

```diff
@@ -115,3 +123,48 @@ func (i ProjectInvitationStatus) Validate() error {
 
 	return errProjectInvitationStatusInvalid
 }
+
+func (r ProjectRole) IsRoleAtLeast(role ProjectRole) bool {
+	return projectRolePriority[r] <= projectRolePriority[role]
+}
+
+type ProjectMember struct {
+	UserID      uuid.UUID
+	ProjectID   uuid.UUID
+	ProjectRole ProjectRole
+	UserData    UserData
```

Comment:
I think it would be better to avoid tightly coupling the project member with the user. Instead of having a User as part of the ProjectMember struct, I would only include `UserID` and `UserData` (user email and name)